### PR TITLE
fix(logging): review and adjust log message and level in other controllers

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#5888](https://github.com/longhorn/longhorn/issues/5888)

### CODE REVIEW

File: controller/volume_controller.go [hunk 1]
```diff
@@ -3783,7 +3776,7 @@ func (c *VolumeController) processMigration(v *longhorn.Volume, es map[string]*l
 	invalidMigrationEngine := availableEngines > 0 && migrationEngine == nil
 	if unexpectedEngineCount || invalidMigrationEngine {
 		revertRequired = true
-		log.Errorf("unexpected state for migration, current engine count %v has invalid migration engine %v",
+		log.Warnf("unexpected state for migration, current engine count %v has invalid migration engine %v",
```
Comment (innobead) [inline controller/volume_controller.go:3779]: `Unexpected ..`

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Some comments need to resolve.

File: controller/share_manager_controller.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (c *ShareManagerController) enqueueShareManagerForVolume(obj interface{}) {
 	if volume.Spec.AccessMode == longhorn.AccessModeReadWriteMany && !volume.Spec.Migratable {
 		// we can queue the key directly since a share manager only manages a single volume from it's own namespace
 		// and there is no need for us to retrieve the whole object, since we already know the volume name
-		getLoggerForVolume(c.logger, volume).Trace("Enqueuing share manager for volume")
+		getLoggerForVolume(c.logger, volume).Info("Enqueuing share manager for volume")
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/share_manager_controller.go:149]: removed, thanks!

File: controller/share_manager_controller.go [hunk 3]
```diff
@@ -173,7 +173,7 @@ func (c *ShareManagerController) enqueueShareManagerForPod(obj interface{}) {
 	// we can queue the key directly since a share manager only manages pods from it's own namespace
 	// and there is no need for us to retrieve the whole object, since the share manager name is stored in the label
 	smName := pod.Labels[types.GetLonghornLabelKey(types.LonghornLabelShareManager)]
-	c.logger.WithField("pod", pod.Name).WithField("shareManager", smName).Trace("Enqueuing share manager for pod")
+	c.logger.WithField("pod", pod.Name).WithField("shareManager", smName).Info("Enqueuing share manager for pod")
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/share_manager_controller.go:176]: removed

File: controller/share_manager_controller.go [hunk 4]
```diff
@@ -354,7 +350,7 @@ func (c *ShareManagerController) syncShareManagerEndpoint(sm *longhorn.ShareMana
 	}
 
 	if service == nil {
-		c.logger.Warn("missing service for share-manager, unsetting endpoint")
+		c.logger.Warn("Unsetting endpoint due to missing service for share-manager")
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/share_manager_controller.go:353]: Fixed,

File: controller/share_manager_controller.go [hunk 5]
```diff
@@ -459,13 +455,12 @@ func (c *ShareManagerController) syncShareManagerVolume(sm *longhorn.ShareManage
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			if sm.Status.State != longhorn.ShareManagerStateStopped {
-				getLoggerForShareManager(c.logger, sm).Infof("stopping share manager because the volume %v is not found", sm.Name)
+				getLoggerForShareManager(c.logger, sm).Infof("Stopping share manager because the volume %v is not found", sm.Name)
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/share_manager_controller.go:458]: Fixed

File: controller/volume_controller.go [hunk 6]
```diff
@@ -1854,7 +1852,6 @@ func (c *VolumeController) canInstanceManagerLaunchReplica(r *longhorn.Replica)
 	}
 	defaultIM, err := c.ds.GetInstanceManagerByInstance(r)
 	if err != nil {
-		return false, errors.Wrapf(err, "cannot find instance manager for replica %v", r.Name)
 	}
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/volume_controller.go:1855]: Fixed.

File: controller/volume_controller.go (hunk 1 above)
Comment (ChanYiLin) [reply controller/volume_controller.go:3779]: Fixed

**Review by innobead — APPROVED** [review]
LGTM.

@ChanYiLin although we merged all changes here, need to continue following up on changed logging, or see if anything is missed to change. Also, community feedback matters, so ensure our logging convention does make sense. Change the convention if needed later. 

BTW, need to create or leverage a validation way later to reduce tedious manual review.

**Comment by innobead** [comment]
@mergify backport v1.5.x

**Pull request merged**